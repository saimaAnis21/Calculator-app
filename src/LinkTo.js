import React from 'react';
import PropTypes from 'prop-types';

const LinkTo = (props) => {
  const { name, path } = props;
  return (
    <a className="" href={path}>{name}</a>
  );
};

LinkTo.propTypes = {
  name: PropTypes.string.isRequired,
  path: PropTypes.string.isRequired,
};

export default LinkTo;
