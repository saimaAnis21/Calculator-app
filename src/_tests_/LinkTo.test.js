import React from 'react';
import renderer from 'react-test-renderer';
import LinkTo from '../LinkTo';

describe('renders LinkTo correctly', () => {
  it('renders Home link correctly', () => {
    const tree = renderer
      .create(<LinkTo name="Home" path="/" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders Calculator link correctly', () => {
    const tree = renderer
      .create(<LinkTo name="Calculator" path="/App" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders Quote link correctly', () => {
    const tree = renderer
      .create(<LinkTo name="Quote" path="/Quote" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
