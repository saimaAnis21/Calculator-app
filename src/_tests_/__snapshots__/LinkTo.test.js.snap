// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders LinkTo correctly renders Calculator link correctly 1`] = `
<a
  className=""
  href="/App"
>
  Calculator
</a>
`;

exports[`renders LinkTo correctly renders Home link correctly 1`] = `
<a
  className=""
  href="/"
>
  Home
</a>
`;

exports[`renders LinkTo correctly renders Quote link correctly 1`] = `
<a
  className=""
  href="/Quote"
>
  Quote
</a>
`;
